shimmingtoolbox/unwrap/prelude.py

```diff
@@ -64,8 +61,8 @@ def prelude(wrapped_phase, affine, mag=None, mask=None, threshold=None, is_unwra
     # Add mask data and options if there is a mask provided
     if mask is not None:
         if mask.shape != wrapped_phase.shape:
-            raise RuntimeError("Mask must be the same shape as wrapped_phase")
-        nii_mask = nib.Nifti1Image(mask, affine)
+            raise ValueError("Mask must be the same shape as wrapped_phase")
+        nii_mask = nib.Nifti1Image(mask, nii_wrapped_phase.affine, header=nii_wrapped_phase.header)
```

Comment:
Some keys might differ between the phase and magnitude headers (e.g. "description") --> They could be replaced before the creation of the new magnitude nifti object